### ISSUE TITLE
FTP error in LAN Modus

### ISSUE THREAD

**Issue by Knaudler** [bambulab/BambuStudio#3016]
### Bambu Studio Version

1.7.7.89

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 10 latest

### Additional system information

gaming pc, 64gb, 24 cores, ssd, 3080

### Printer

Bambu Lab A1 mini with AMS

### How to reproduce

this model
https://makerworld.com/de/models/69266#profileId-75528
then go to print plate

with smaller models ok
only sent to printer works (then manually print model from sd)
but print on printer doesnt work
same with Bambustudio and with orca slicer

### Actual results

startet to sent to printer 10%...20%...30% then error
error code 4020
failed to upload print file to ftp
estra info: ftp code 9
access denied to remote resource: Server denied you to change to the given diretory

### Expected results

it should print
small prints ok
calibration tests ok
video/cam ok
I formatted the sd-card (in the printer)
I resetted the printer

### Project file & Debug log uploads

[fehler-a1-poop-chute-bottom.zip](https://github.com/bambulab/BambuStudio/files/13537741/fehler-a1-poop-chute-bottom.zip)
[debug_Sat_Dec_02_22_41_04_26348.log.zip](https://github.com/bambulab/BambuStudio/files/13537742/debug_Sat_Dec_02_22_41_04_26348.log.zip)

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by Knaudler** [comment]
ok 
with raw stl it works
so the 3mf makes problem

**Comment by walterwongbbl** [comment]
Hi. @Knaudler 
It seems that the FTP transfer failed, and this error message is not accurate.
I tried to reproduce using the file you provided, and the result was correct. You can use the latest version 1.8 to revalidate.
https://github.com/bambulab/BambuStudio/releases/tag/v01.08.00.62

**Comment by helo-head** [comment]
I'm having the exact same issue on a Mac with exact same error messages. 
OS: Sonoma 14.1.2 
BambuStudio: 1.7.7.89 Printer: X1C with firmware 01.06.05.01
Network: Local Lan Printing

What is interesting is that this worked fine till today, no updates with exception of Mac security fixes,  it just stopped working after 5 months.

**Comment by walterwongbbl** [comment]
@helo-head 
Is it the same file? How about changing to a smaller file. In addition, the LAN functionality will have much stricter requirements for the network.

**Comment by helo-head** [comment]
Exact same file. Over three days I tried several 3mf and STL files and in all cases I saw the same behavior in that I was able to "Send" the file successfully while unable to "Print Plate". 

Unfortunately or fortunately depending on your view point my problem has resolved itself this morning before I got a chance to sniff the network. AS such Im not sure I can provide any more insight unless the problem comes back.

**Comment by hxdaro** [comment]
I am having this same error. I have rebooted the printer, rebooted my computer, reinstalled Bambu Studio..

I can send GCode to the printer with the "SEND ALL" command successfully, but trying to use the print plate/print all button causes the FTP error.

**Comment by hxdaro** [comment]
@walterwongbbl ^

**Comment by walterwongbbl** [comment]
@hxdaro 
I think this may be related to the FTP code we are using, and we are trying to optimize it

**Comment by kool-de** [comment]
Same issue on my MAC.
OS: Sonoma 14.2.1
BambuStudio: 1.8.2.56
Printer: X1C with firmware 01.07.02.00
Network: Local Lan Printing

Send-function works, Print everytime aborts. Macbook reboot and 3D-printer reboot does not solve the issue.
Uploading files with FTP-client(FTP over TLS) works fine without any problems.

**Comment by 459below** [comment]
The error occurred for me in the same manner as described by @Knaudler 

It looks to me the provided SDcard might have just died. After I worked around the FTP transfer error, by converting 3MF to STL and print from there, it stopped dead after 10% of printing. BambuStudio said
```
MicroSD Card read/write exception. please reinsert or replace MicroSD Card. [0500 C010 15:12:54]
```

The MicroSD cards controller seems to have shut down presenting the card only on read-only mode. I know this from USB sticks that have died after too much use. I will be replacing the sdcard now and move on.

Information:

LAN only mode: Yes
Printer: P1S
Firmware: 01.05.02.00 (20240122)
Bambu Studio Version: 01.08.04.51 (Flatpak)
3MF file BambuStudio has issue with: https://www.printables.com/model/503881-funnel-for-adjustable-spool-holder-for-silica-gel-/files
Related topic: https://forum.bambulab.com/t/updated-local-printing-now-failing-with-ftp-access-denied-errors/39853

**Comment by 459below** [comment]
So it turns out even after swapping the SD card (MBR, FAT32, btw) the FTP error still occurred. So the workaround was:

- Open 3MF file in BambuStudio
- Export it as STL
- Create a new project in BambuStudio
- Import the same STL
- Print as usual

As there are several other customers having this particular issue with the closed-source firmware, I think this issue should be reopened until the FTP code to transfer 3MF files has been fixed.

---

As for the topic of the SD card dying I disabled the recording in the Smartphone app "Settings -> Record Settings -> Videos" and shortly before the print in BambuStudio "Time Lapse"

Analyzing the space usage of the already died SD card I noticed that the folders _cam, ipcam and timelapse_ holding video files made up 8 GB of space, while everything else including 24 different gcodes and 3mfs was sitting at only 0.1 GB.

**Comment by eolder** [comment]
> the workaround was:
> 
>     * Open 3MF file in BambuStudio
> 
>     * Export it as STL
> 
>     * Create a new project in BambuStudio
> 
>     * Import the same STL
> 
>     * Print as usual
> As there are several other customers having this particular issue with the closed-source firmware, I think this issue should be reopened until the FTP code to transfer 3MF files has been fixed.

I can confirm replicating this behavior. I've only ever had it occur when using a 3MF from Makerworld.

**Comment by skintigh** [comment]
According to the log file there are numbers and dashes in your filename. BambuLabs can't handle anything other than lowercase letters it seems; try renaming the file.

I just tested this with "ring0.stl" then renamed it to "ringzero.stl" and the first failed at 70% with an FTP error while the second printed perfectly. 

Not one byte was different, I made the exact same clicks when importing both files. The only difference was I changed the filename in windows before the second print. It seems including the number 0 somehow resulting in an interrupted connection and then the printer does crazy things to reestablish it that get blocked by Windows Defender. Turning the firewall off allows me to print "ring0.stl"
